### DISCUSSION TITLE
How can I avoid that editor toolbar buttons are disappearing in my theme?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#988] [Ask a question] [answered]
While helping a friend to create a new website, I noticed that the toolbar buttons in the editor were not complete, even though the setting was set to auto. For example, the status button and the page preview were missing. Apparently this is due to my [Wittstock theme](https://github.com/schulle4u/yellow-wittstock), because in the default Stockholm theme everything works as expected. I can reproduce this behaviour in a new installation. As soon as Wittstock is installed and set as default theme, some toolbar buttons are missing. I never noticed this myself, as I rarely need the buttons, but my friend does 🙂 With a little experimentation I found out that changing the page zoom affects the behavior of the toolbar buttons. The smaller the page, the fewer buttons, but some buttons also disappear at higher zoom levels above 100%. In my theme, however, I can't get all the buttons to appear, regardless of the zoom level. I roughly understand why the buttons behave this way, but I am not sure which CSS settings or even HTML elements affect how the toolbar fits into the page. Does any of the CSS experts have an idea?

**Comment by xrat** [comment]
I am interested too. I haven't been digging as deep as you since opting to make the body a bit wider helped:

``` css
body { grid-template-columns: 1fr min(49rem, 90%) 1fr; }
```

**Reply by schulle4u** [reply, marked as answer]
Thanks, widening the second grid column seems to be the best solution here. Well, I should finally make a cheat sheet so I can keep track of all the changes from the original theme when I update `simple.css` from upstream in the future. 😅 

It seems that all themes have the problem that the toolbar may not be completely displayed when the zoom level is changed, although this behavior is probably intentional so as not to destroy the entire layout.

**Comment by auer-kk** [comment]
The dimensions of the edit-pane are calculated by javascript. If you make the following changes to `wittstock.css`the width of the inserted div `.yellow-bar` will be the whole width of `body`.

```
body > *:not(.yellow-bar) {
  grid-column: 2;
}

body > .yellow-bar {
  grid-column: 1 / -1;
}
```

**Reply by schulle4u** [reply]
Thank you for the suggestion. I've added it to my Wittstock theme as a more elegant solution which leaves the remaining styles untouched. 👍

**Comment by annaesvensson** [comment]
What you describe is the intended behaviour of a responsive user interface. The [built-in web editor](https://github.com/annaesvensson/yellow-edit) displays as many toolbar buttons as will fit in one row, if there is not enough space then fewer buttons will be displayed. The available width is calculated using JavaScript. You can customise the available width with the help of CSS, but ultimately the editor determines how many buttons are displayed when you resize the web browser. Perhaps the responsive user interface can be improved in the future? Perhaps something like <kbd>...</kbd> or <kbd>&gt;&gt;&gt;</kbd> or a kind of "menu" button to access the disappearing buttons?

**Reply by schulle4u** [reply]
Yes, a good idea to try out, as GitHub also implements this in their editor. Maybe the editor even could have a distraction-free mode and open the whole toolbar in a sidebar? Otherwise I think the current implementation is also fine. You just have to get used to the behavior of the editor as a theme developer.

**Reply by annaesvensson** [reply]
I think the web editor should work well with a variety of themes, devices and screen sizes. Let us check how much source code it would take to implement, I'll get back to you.

**Reply by annaesvensson** [reply]
A reminder that this is still open. 😉

**Comment by pawell0** [comment]
I have also a problem using the Stockholm theme. The buttons in the editing panel (the ones from “Format” to “Preview”) disappear behind the ‘Cancel" and “Save” buttons when I'm editing content on my phone. 😞 It's happening gradually depending on the screen resolution.  Editing a webpage on a phone is very inconvenient with this. In my particular Yellow use it’s very annoying.

<img width="308" height="308" alt="hiding" src="https://github.com/user-attachments/assets/03dba56e-2618-4e97-ad04-005996a5bdcd" />


But why not place the “Cancel” and ‘Save’ buttons to the left of the “Format” button? That way, all the buttons would be in a single row, or two rows, if the window is narrow. Is it really necessary to keep the buttons in a single row?

**Reply by pftnhr** [reply]
Looks like, you're on a mobile device. If you turn it to landscape mode you should see all buttons.

**Reply by pawell0** [reply]
Of course. I mentioned editing on a phone in my post. Constantly having to edit and rotate the screen is very poor ergonomics, especially when the on-screen keyboard blocks a large part of your view while you're editing. That’s why something needs to be done to ensure these buttons are always visible in a portrait mode, regardless of how many there are.

**Reply by pftnhr** [reply]
Yes that's right. It also happens on the desktop if your theme is less wider. So I made the class `.yellow-bar` full width.
